libmount/src/context_mount.c

```diff
@@ -1068,6 +1138,97 @@ static int parse_ownership_mode(struct libmnt_context *cxt)
 	return 0;
 }
 
+#ifdef UL_HAVE_MOUNT_API
+/*
+ * Process X-mount.idmap=.
+ */
+static int mnt_context_prepare_idmaps(struct libmnt_context *cxt)
+{
+	int rc;
+	char *value;
+	size_t valsz;
+	char *saveptr = NULL, *tok;
+
+	const char *o = mnt_fs_get_user_options(cxt->fs);
+	if (!o)
+		return 0;
+
+	rc = mnt_optstr_get_option(o, "X-mount.idmap", &value, &valsz);
+	if (rc < 0)
+		return -MNT_ERR_MOUNTOPT;
+	if (rc > 0)
+		return 0;
+
+	if (!valsz)
+		return errno = EINVAL, -MNT_ERR_MOUNTOPT;
+
+	/* Has the user given us a path to a user namespace? */
+	if (startswith(value, "/")) {
```

Comment:
if (*value == '/')

:-)

Reply:
Yeah, I thought about that but then thought maybe I should use the helper. :) Will switch!